### ISSUE TITLE
Params in HAconv

### ISSUE THREAD

**Issue by huixiancheng** [shachoi/HANet#19]
Hello! I would like to know if the number **128 in HAconv and PosEncoding1D** corresponds to the **128 in pos_h**? 
https://github.com/shachoi/HANet/blob/9c31ef21df1ee471636d1de7e3af06f959315e5b/network/PosEmbedding.py#L55-L64
https://github.com/shachoi/HANet/blob/9c31ef21df1ee471636d1de7e3af06f959315e5b/network/HANet.py#L51-L56
https://github.com/shachoi/HANet/blob/9c31ef21df1ee471636d1de7e3af06f959315e5b/datasets/cityscapes.py#L740
Add if I set pos_h to 64 (like under) , should the correspondence 128 be changed to 64 in HAconv and PosEncoding1D?
~~~
self.pos_h = torch.arange(0, 64).unsqueeze(0).unsqueeze(2).expand(-1,-1,2048)  # 0 ~ 127
~~~

Also, Look like you set two optims and schedulers for model'params(optimizer, scheduler) and hanet's params(optimizer_at, scheduler_at ). But in terms of config, they are the same except for weight_decay. Why is this？ Also is it necessary to set different weight_deacy  for the HA module?

**Comment by sunghac** [comment]
First of all, my code was too messy :( Unnecessary code generated through various experiments remains.
1. If you want to change the \hat{H}, just modify pos_rfactor. Default value of \hat{H} is 16 (=128//(pos_rfactor(=8)))

<img width="388" alt="image" src="https://user-images.githubusercontent.com/34509882/166140509-b6257613-772b-455c-899c-27ca888b29c2.png">

2. As you said, optimizers can be integrated into one, not separate. Different weight decay for HANet is not a big deal. In another experiment, I tried to optimize the HANet part with a separate loss, and a separate optimizer seems an unnecessary remnant.

I'll try to clean up the code later when I have time.

Thank you!
Sungha.

**Comment by huixiancheng** [comment]
Hi. Thanks for your reply! 
Yes, position information is initially set to from 0 to 127, and it is reduced to from 0 to 15 using the 8 as pos_rfactor. Change the pos_rfactor could change the \hat{H}. 
However, the key problem is that my inputs couldn't be initialized with a position information from 0 to 127. The resolution of my pic is low resolution on height (such as 64*512). The initialized location information should be represented from 0 to 63 (may be I'm wrong on this). So in this case, shou I change all the 128 to 64 in HAconv and PosEncoding?

**Comment by sunghac** [comment]
Yes, that's right. you already understand well.
You can modify pos_h as you mentioned, and you should change all the 128 to 64 in HANet and PosEncoding.
However, it seems to be necessary to find the optimal reduction factor through experiments.
You may not need noise, and depending on the nature of your image you may not need to reduce the height.
```
self.noise = torch.distributions.normal.Normal(torch.tensor([0.0]), torch.tensor([pos_noise])) 
```

**Comment by huixiancheng** [comment]
Thank you for your kind reply and advice. :bow:  
I will take a try.
